### DISCUSSION TITLE
Understanding composeContextScopes

### DISCUSSION THREAD

**Discussion by jd-carroll** [radix-ui/primitives#2304] [Help]
My most fundamental question is probably somewhere along the lines of "Can I use `CollapsibleContent` independent of a parent `Collapsible`?" More broadly I've encountered a few situations where I've wanted to separate the "trigger" from the "primitive" but (being a newb) not sure I've fully grasped the best way to do that.

However, I think fundamental to all of that is the whole `createContextScope`, `createContext`, and `use[Composed]Context` pattern.

Does anyone have an explainer they could point me to to better understand what is going on with that pattern? (Any tips for how to use the `createCollapsibleScope` like a pro?)

**Comment by joaom00** [comment]
the `contextScope` stuff is for internal use only. But if you want to understand more deeply what it does you can read this [thread](https://github.com/radix-ui/primitives/discussions/1091#discussioncomment-1984109)

> I've wanted to separate the "trigger" from the "primitive"

Not sure I follow, can you explain?

**Comment by jd-carroll** [comment]
I am probably just hung up on "how its done in the docs" versus how you could do it.  Maybe a better example would be a dialog.

It's great that you can declare a dialog like this:
```jsx
  <Dialog.Root>
    <Dialog.Trigger asChild>
      <button className="Button violet">Edit profile</button>
    </Dialog.Trigger>
    <Dialog.Portal>
      <Dialog.Overlay className="DialogOverlay" />
      <Dialog.Content className="DialogContent">
```

_(Where the button trigger is declared under the dialog.)_

But that is just for convenience and there is nothing wrong with passing `open` and `onOpenChange` to the dialog instead.
